### DISCUSSION TITLE
Switching GOSPA

### DISCUSSION THREAD

**Discussion by Carlson-J** [dstl/Stone-Soup#876] [Q&A] [answered]
Is there the capability to add a switching term for the GOSPA metric? As done in https://www.mathworks.com/help/fusion/ref/trackgospametric-system-object.html#d126e213697, having a term that penalizes when a predicted track switches between two truth tracks would be very useful.

I'm new to Stone-Soup so if this is already a capability or something that is easily done with the framework please let me know.

**Comment by sdhiscocks** [comment, marked as answer]
This isn't something currently implemented. Taking a quick look at the code, it may not be too difficult to implement.

This following line is executed for each timestep, and includes the `truth_to_measured_assignment`. This could be stored and compared to the previous timestep to see if switch occurred, but would have to be careful to spot cases where tracks or targets died.
https://github.com/dstl/Stone-Soup/blob/71b47f9c19bfdbc225d659b34277502052b4e4e0/stonesoup/metricgenerator/ospametric.py#L129
